### ISSUE TITLE
New dev container hangs and never completes

### ISSUE THREAD

**Issue by evilhamsterman** [microsoft/vscode-remote-release#9892]
Trying to just create a dev container without an associated folder or repo hangs. 

- VSCode Version: 1.89.1 and Insiders 1.90.0
- Local OS Version: Windows 11 Pro 23H2
- Remote OS Version: NA
- Remote Extension/Connection Type: Containers

- Logs:
```
[692 ms] Dev Containers 0.366.0 in VS Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685).
[691 ms] Initializing configuration support...
[692 ms] Internal initialization of dev container support package...
[3534 ms] Start: Run: docker version --format {{json .}}
[3584 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[3586 ms] Start: Run: docker -v
[3634 ms] Start: Run: docker events --format {{json .}} --filter event=create --filter event=destroy --filter event=die --filter event=health-status --filter event=kill --filter event=pause --filter event=rename --filter event=resize --filter event=restart --filter event=start --filter event=stop --filter event=unpause --filter event=update --filter event=exec_create
[3647 ms] Start: Run: docker ps -q -a
[3648 ms] Start: Run: docker version --format {{json .}}
[3705 ms] Start: Run: docker inspect --type container 6acd0132193b
[3706 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[3708 ms] Start: Run: docker -v
[3755 ms] Start: Run: docker events --format {{json .}} --filter event=destroy --filter event=die --filter event=kill --filter event=create --filter event=rename --filter event=resize --filter event=restart --filter event=start --filter event=stop --filter event=unpause --filter event=prune
[3768 ms] Start: Run: docker volume ls -q
[3825 ms] Start: Run: docker inspect --type volume tailsshale-main-484e6f0ce03c9473996b81a058111c50fb54923dbae3e422e91385e708994cd4 ubuntu
[5416 ms] Start: Run: docker inspect --type container 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[6571 ms] Start: Run: docker inspect --type container 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[6620 ms] Start: Run: docker stop 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[6674 ms] Start: Run: docker inspect --type container 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[8007 ms] Start: Run: docker ps -q -a
[8056 ms] Start: Run: docker inspect --type container 6acd0132193b
[16682 ms] Start: Run: docker inspect --type container 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[17106 ms] Start: Run: docker inspect --type container 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[17107 ms] Start: Run: docker rm 6acd0132193b6776d85de8e5e4de1800bf2413f4a7fa02180c18938b853c3b54
[17111 ms] Start: Run: docker volume ls -q
[17164 ms] Start: Run: docker inspect --type volume tailsshale-main-484e6f0ce03c9473996b81a058111c50fb54923dbae3e422e91385e708994cd4 ubuntu
[17180 ms] Start: Run: docker ps -q -a
[17385 ms] Start: Run: docker volume ls -q
[17387 ms] Start: Run: docker ps -q -a
[17434 ms] Start: Run: docker inspect --type volume tailsshale-main-484e6f0ce03c9473996b81a058111c50fb54923dbae3e422e91385e708994cd4 ubuntu
[87961 ms] Start: Run: docker volume ls -q -f dangling=true -f label=vsch.local.repository
[88006 ms] Start: Run: docker volume ls -q -f dangling=true -f label=dev.container.volume
[88051 ms] Start: Run: docker inspect --type volume tailsshale-main-484e6f0ce03c9473996b81a058111c50fb54923dbae3e422e91385e708994cd4 ubuntu
[90174 ms] Start: Run: docker volume rm tailsshale-main-484e6f0ce03c9473996b81a058111c50fb54923dbae3e422e91385e708994cd4 ubuntu
[90220 ms] Start: Run: docker volume ls -q
[90221 ms] Start: Run: docker ps -q -a
[90423 ms] Start: Run: docker ps -q -a
[90424 ms] Start: Run: docker volume ls -q
[94504 ms] Start: Run: docker version --format {{json .}}
[94550 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[94552 ms] Start: Run: docker pull alpine:3.16.3
[94822 ms] Start: Run: docker volume ls -q
[97794 ms] Start: Run: docker volume create --label dev.container.volume=true ubuntu
[97841 ms] Start: Run: docker volume ls -q
[97842 ms] Start: Run: docker ps -q -a
[97843 ms] Start: Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[97934 ms] Start: Run: docker inspect --type volume ubuntu
[98146 ms] Start: Run: docker ps -q -a
[98146 ms] Start: Run: docker volume ls -q
[98194 ms] Start: Run: docker inspect --type volume ubuntu
[98211 ms] Start: Run: docker inspect --type container a5fcf67bc71a
[98321 ms] Preparing to apply Template 'ghcr.io/devcontainers/templates/ubuntu:latest'
[98323 ms] Running Dev Containers CLI:   templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1715903376522 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1715903376522 --log-level debug
[98326 ms] ProxyResolver#loadSystemCertificates count
[98328 ms] ProxyResolver#loadSystemCertificates count filtered
[98329 ms] Start: Run: C:\Users\evilh\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\evilh\.vscode\extensions\ms-vscode-remote.remote-containers-0.366.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1715903376522 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1715903376522 --log-level debug
[98531 ms] [2024-05-16T23:49:36.730Z] @devcontainers/cli 0.60.0. Node.js v18.18.2. win32 10.0.22631 x64.
[98536 ms] [2024-05-16T23:49:36.735Z] Loading 50 extra certificates from C:\Users\evilh\AppData\Local\Temp\vsch\certificates-9b3207a1811c345e81b95e721f2c3bebc32924a0050cf938fcb28057258a1684.pem.
[98702 ms] [2024-05-16T23:49:36.901Z] Start: Run: docker-credential-wincred get
[98730 ms] [2024-05-16T23:49:36.929Z] Stop (28 ms): Run: docker-credential-wincred get
[99290 ms] [2024-05-16T23:49:37.489Z] Replacing imageVariant with jammy
```

Steps to Reproduce:

1. Open the command pallete and run `Dev Containers: New Dev Container`
2. Select any template, for example Ubuntu
3. Press Enter and watch it hit the logs above and then just do nothing. I used a fresh install of Rancher Desktop with and without Kubernetes installed. Same thing happens on release and a fresh insiders install with only the prerelease dev containers extension.

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by chrmarti** [comment]
Could you change the Dev Containers' log level in the user settings to `trace` and retry:
![Image](https://github.com/microsoft/vscode-remote-release/assets/9205389/26a850af-7224-451a-b591-a2b966ab0f14)

**Comment by chrmarti** [comment]
Related: https://github.com/devcontainers/templates/issues/242

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by evilhamsterman** [comment]
Sorry I was unavailable the last couple weeks. I enabled trace logging it doesn't provide much more info 
```
[526 ms] Dev Containers 0.370.0 in VS Code 1.90.0 (89de5a8d4d6205e5b11647eb6a74844ca23d2573).
[525 ms] Initializing configuration support...
[526 ms] Internal initialization of dev container support package...
[4494 ms] Start: Run: docker version --format {{json .}}
[4546 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[4549 ms] Start: Run: docker -v
[4600 ms] Start: Run: docker events --format {{json .}} --filter event=create --filter event=destroy --filter event=die --filter event=health-status --filter event=kill --filter event=pause --filter event=rename --filter event=resize --filter event=restart --filter event=start --filter event=stop --filter event=unpause --filter event=update --filter event=exec_create
[4612 ms] Start: Run: docker ps -q -a
[4624 ms] Start: Run: docker version --format {{json .}}
[4670 ms] Start: Run: docker inspect --type container 25032e6c4944 00bf97e35d6e
[4676 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[4678 ms] Start: Run: docker -v
[4730 ms] Start: Run: docker events --format {{json .}} --filter event=destroy --filter event=die --filter event=kill --filter event=create --filter event=rename --filter event=resize --filter event=restart --filter event=start --filter event=stop --filter event=unpause --filter event=prune
[4743 ms] Start: Run: docker volume ls -q
[4790 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e go vscode
[6289 ms] Start: Run: docker inspect --type container 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[7709 ms] Start: Run: docker inspect --type container 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[7759 ms] Start: Run: docker stop 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[7811 ms] Start: Run: docker inspect --type container 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[9881 ms] Start: Run: docker ps -q -a
[9929 ms] Start: Run: docker inspect --type container 25032e6c4944 00bf97e35d6e
[17821 ms] Start: Run: docker inspect --type container 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[18252 ms] Start: Run: docker inspect --type container 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[18255 ms] Start: Run: docker rm 25032e6c4944ba5e94291e7125602e0e6d01f183280812891646b64e951f922d
[18257 ms] Start: Run: docker volume ls -q
[18325 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e go vscode
[18327 ms] Start: Run: docker ps -q -a
[18379 ms] Start: Run: docker inspect --type container 00bf97e35d6e
[18518 ms] Start: Run: docker volume ls -q
[18575 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e go vscode
[18655 ms] Start: Run: docker ps -q -a
[18715 ms] Start: Run: docker inspect --type container 00bf97e35d6e
[20682 ms] Start: Run: docker volume ls -q -f dangling=true -f label=vsch.local.repository
[20729 ms] Start: Run: docker volume ls -q -f dangling=true -f label=dev.container.volume
[20774 ms] Start: Run: docker inspect --type volume go
[23503 ms] Start: Run: docker volume rm go
[23549 ms] Start: Run: docker volume ls -q
[23551 ms] Start: Run: docker ps -q -a
[23598 ms] Start: Run: docker inspect --type container 00bf97e35d6e
[23598 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e vscode
[46880 ms] Start: Run: docker version --format {{json .}}
[46926 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[46927 ms] Stop (47 ms): Run: docker version --format {{json .}}
[46930 ms] Start: Run: docker pull alpine:3.16.3
[47099 ms] Start: Run: docker volume ls -q
[47145 ms] Stop (46 ms): Run: docker volume ls -q
[53652 ms] Start: Run: docker volume create --label dev.container.volume=true ubuntu
[53700 ms] Start: Run: docker ps -q -a
[53701 ms] Start: Run: docker volume ls -q
[53701 ms] Stop (49 ms): Run: docker volume create --label dev.container.volume=true ubuntu
[53702 ms] Start: Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[53754 ms] Stop (54 ms): Run: docker ps -q -a
[53755 ms] Start: Run: docker inspect --type container 00bf97e35d6e
[53755 ms] Stop (54 ms): Run: docker volume ls -q
[53755 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode
[53807 ms] Stop (52 ms): Run: docker inspect --type container 00bf97e35d6e
[53807 ms] Stop (52 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode
[53980 ms] Stop (278 ms): Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[53986 ms] Start: Run: docker volume ls -q
[54020 ms] Start: Run: docker ps -q -a
[54035 ms] Stop (49 ms): Run: docker volume ls -q
[54035 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode
[54061 ms] Preparing to apply Template 'ghcr.io/devcontainers/templates/ubuntu:latest'
[54066 ms] Running Dev Containers CLI:   templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1717631745405 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1717631745405 --log-level trace
[54070 ms] ProxyResolver#loadSystemCertificates count
[54072 ms] ProxyResolver#loadSystemCertificates count filtered
[54073 ms] Start: Run: C:\Users\evilh\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\evilh\.vscode\extensions\ms-vscode-remote.remote-containers-0.370.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1717631745405 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1717631745405 --log-level trace
[54081 ms] Stop (61 ms): Run: docker ps -q -a
[54081 ms] Start: Run: docker inspect --type container 1b16d3e9bd92 00bf97e35d6e
[54095 ms] Stop (60 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode
[54144 ms] Stop (63 ms): Run: docker inspect --type container 1b16d3e9bd92 00bf97e35d6e
[54273 ms] [2024-06-05T23:55:45.618Z] @devcontainers/cli 0.62.0. Node.js v20.9.0. win32 10.0.22631 x64.
[54273 ms] [2024-06-05T23:55:45.618Z] > input: ghcr.io/devcontainers/templates/ubuntu:latest
[54273 ms] [2024-06-05T23:55:45.619Z] >
[54274 ms] [2024-06-05T23:55:45.619Z] > resource: ghcr.io/devcontainers/templates/ubuntu
[54274 ms] [2024-06-05T23:55:45.619Z] > id: ubuntu
[54274 ms] [2024-06-05T23:55:45.619Z] > owner: devcontainers
[54274 ms] [2024-06-05T23:55:45.619Z] > namespace: devcontainers/templates
[54274 ms] [2024-06-05T23:55:45.619Z] > registry: ghcr.io
[54274 ms] [2024-06-05T23:55:45.619Z] > path: devcontainers/templates/ubuntu
[54274 ms] [2024-06-05T23:55:45.619Z] >
[54274 ms] [2024-06-05T23:55:45.619Z] > version: latest
[54274 ms] [2024-06-05T23:55:45.619Z] > tag?: latest
[54274 ms] [2024-06-05T23:55:45.619Z] > digest?: undefined
[54276 ms] [2024-06-05T23:55:45.619Z] > input: ghcr.io/devcontainers/templates/ubuntu:latest
[54276 ms] [2024-06-05T23:55:45.619Z] >
[54276 ms] [2024-06-05T23:55:45.619Z] > resource: ghcr.io/devcontainers/templates/ubuntu
[54276 ms] [2024-06-05T23:55:45.619Z] > id: ubuntu
[54276 ms] [2024-06-05T23:55:45.619Z] > owner: devcontainers
[54277 ms] [2024-06-05T23:55:45.619Z] > namespace: devcontainers/templates
[54277 ms] [2024-06-05T23:55:45.619Z] > registry: ghcr.io
[54277 ms] [2024-06-05T23:55:45.619Z] > path: devcontainers/templates/ubuntu
[54277 ms] [2024-06-05T23:55:45.619Z] >
[54277 ms] [2024-06-05T23:55:45.619Z] > version: latest
[54277 ms] [2024-06-05T23:55:45.619Z] > tag?: latest
[54277 ms] [2024-06-05T23:55:45.619Z] > digest?: undefined
[54277 ms] [2024-06-05T23:55:45.619Z] manifest url: https://ghcr.io/v2/devcontainers/templates/ubuntu/manifests/latest
[54279 ms] [2024-06-05T23:55:45.625Z] Loading 50 extra certificates from C:\Users\evilh\AppData\Local\Temp\vsch\certificates-9b3207a1811c345e81b95e721f2c3bebc32924a0050cf938fcb28057258a1684.pem.
[54460 ms] [2024-06-05T23:55:45.806Z] [httpOci] Attempting to authenticate via 'Bearer' auth.
[54462 ms] [2024-06-05T23:55:45.808Z] [httpOci] Invoking credsStore credential helper 'wincred'
[54462 ms] [2024-06-05T23:55:45.808Z] Start: Run: docker-credential-wincred get
[54490 ms] [2024-06-05T23:55:45.836Z] Stop (28 ms): Run: docker-credential-wincred get
[54490 ms] [2024-06-05T23:55:45.836Z] [httpOci] Failed to query for 'ghcr.io' credential from 'docker-credential-wincred': [object Object]
[54490 ms] [2024-06-05T23:55:45.836Z] [httpOci] No authentication credentials found for registry 'ghcr.io' via docker config or credential helper.
[54490 ms] [2024-06-05T23:55:45.836Z] [httpOci] No authentication credentials found for registry 'ghcr.io'. Accessing anonymously.
[54491 ms] [2024-06-05T23:55:45.836Z] [httpOci] Attempting to fetch bearer token from:  https://ghcr.io/token?service=ghcr.io&scope=repository:devcontainers/templates/ubuntu:pull
[54789 ms] [2024-06-05T23:55:46.135Z] [httpOci] 200 on reattempt after auth: https://ghcr.io/v2/devcontainers/templates/ubuntu/manifests/latest
[54790 ms] [2024-06-05T23:55:46.135Z] blob url: https://ghcr.io/v2/devcontainers/templates/ubuntu/blobs/sha256:61ed91d23448484ccf23eae27ac81cdba43bf28ee194b489c9fe02c90b22c8db
[54790 ms] [2024-06-05T23:55:46.136Z] [httpOci] Applying cachedAuthHeader for registry ghcr.io...
[55051 ms] [2024-06-05T23:55:46.396Z] [httpOci] 200 (Cached): https://ghcr.io/v2/devcontainers/templates/ubuntu/blobs/sha256:61ed91d23448484ccf23eae27ac81cdba43bf28ee194b489c9fe02c90b22c8db
[55059 ms] [2024-06-05T23:55:46.405Z] ./ : Directory
[55060 ms] [2024-06-05T23:55:46.405Z] ./.devcontainer/ : Directory
[55060 ms] [2024-06-05T23:55:46.406Z] ./.github/ : Directory
[55060 ms] [2024-06-05T23:55:46.406Z] Skipping file './NOTES.md' during blob extraction
[55060 ms] [2024-06-05T23:55:46.406Z] Skipping file './README.md' during blob extraction
[55060 ms] [2024-06-05T23:55:46.406Z] Skipping file './devcontainer-template.json' during blob extraction
[55060 ms] [2024-06-05T23:55:46.406Z] ./.devcontainer/devcontainer.json : File
[55061 ms] [2024-06-05T23:55:46.406Z] ./.github/dependabot.yml : File
[55064 ms] [2024-06-05T23:55:46.410Z] Files extracted from blob: ./.devcontainer/devcontainer.json, ./.github/dependabot.yml
[55065 ms] [2024-06-05T23:55:46.411Z] Found metadata file 'devcontainer-template.json' in blob
[55071 ms] [2024-06-05T23:55:46.417Z] Using default value for imageVariant --> jammy
[55071 ms] [2024-06-05T23:55:46.417Z] Scanning file './.devcontainer/devcontainer.json'
[55079 ms] [2024-06-05T23:55:46.423Z] Replacing imageVariant with jammy
[55079 ms] [2024-06-05T23:55:46.423Z] Scanning file './.github/dependabot.yml'
[55087 ms] Stop (1014 ms): Run: C:\Users\evilh\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\evilh\.vscode\extensions\ms-vscode-remote.remote-containers-0.370.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1717631745405 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1717631745405 --log-level trace
[55087 ms] {
    "files": [
        "./.devcontainer/devcontainer.json",
        "./.github/dependabot.yml"
    ]
}
[55087 ms] Dev Containers CLI completed
[56078 ms] Stop (9148 ms): Run: docker pull alpine:3.16.3
```

**Comment by evilhamsterman** [comment]
@chrmarti I don't know if you can reopen this or should I file a new issue?

**Comment by chrmarti** [comment]
Not sure what could go wrong here. I'm adding more trace logging to 0.371.0-pre-release. Could you give that a try and post the log again?

**Comment by evilhamsterman** [comment]
I will as soon as I can. Some more information I'm using Rancher Desktop as my Docker

**Comment by evilhamsterman** [comment]
Not much more

```
[620 ms] Dev Containers 0.371.0 in VS Code 1.90.0 (89de5a8d4d6205e5b11647eb6a74844ca23d2573).
[619 ms] Initializing configuration support...
[620 ms] Internal initialization of dev container support package...
[623 ms] Start: Run: docker version --format {{json .}}
[700 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[703 ms] Stop (80 ms): Run: docker version --format {{json .}}
[705 ms] Start: Run: docker -v
[767 ms] Stop (62 ms): Run: docker -v
[767 ms] Start: Run: docker events --format {{json .}} --filter event=create --filter event=destroy --filter event=die --filter event=health-status --filter event=kill --filter event=pause --filter event=rename --filter event=resize --filter event=restart --filter event=start --filter event=stop --filter event=unpause --filter event=update --filter event=exec_create
[782 ms] Start: Run: docker ps -q -a
[861 ms] Stop (79 ms): Run: docker ps -q -a
[1019 ms] Start: Run: docker version --format {{json .}}
[1023 ms] Start: Run: docker inspect --type container 63ded8e7122e 0de6f7091fdb 00bf97e35d6e
[1077 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[1079 ms] Stop (60 ms): Run: docker version --format {{json .}}
[1080 ms] Start: Run: docker -v
[1083 ms] Stop (60 ms): Run: docker inspect --type container 63ded8e7122e 0de6f7091fdb 00bf97e35d6e
[1168 ms] Stop (88 ms): Run: docker -v
[1168 ms] Start: Run: docker events --format {{json .}} --filter event=destroy --filter event=die --filter event=kill --filter event=create --filter event=rename --filter event=resize --filter event=restart --filter event=start --filter event=stop --filter event=unpause --filter event=prune
[1221 ms] Start: Run: docker volume ls -q
[1283 ms] Stop (62 ms): Run: docker volume ls -q
[1284 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[1355 ms] Stop (71 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[3183 ms] Start: Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[3232 ms] Stop (49 ms): Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[4421 ms] Start: Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[4473 ms] Stop (52 ms): Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[4473 ms] Start: Run: docker stop 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[4531 ms] Start: Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[4582 ms] Stop (51 ms): Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14540 ms] Start: Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14796 ms] Start: Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14797 ms] Stop (10324 ms): Run: docker stop 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14798 ms] Start: Run: docker rm 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14800 ms] Start: Run: docker volume ls -q
[14802 ms] Stop (262 ms): Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14848 ms] Stop (52 ms): Run: docker inspect --type container 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14851 ms] Stop (51 ms): Run: docker volume ls -q
[14852 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[14854 ms] Stop (56 ms): Run: docker rm 63ded8e7122ed5d0afda04056c14b0dfb33774a8850910b7055871944c14aa36
[14864 ms] Start: Run: docker ps -q -a
[14906 ms] Stop (54 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[14913 ms] Stop (49 ms): Run: docker ps -q -a
[14913 ms] Start: Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[14963 ms] Stop (50 ms): Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[15064 ms] Start: Run: docker volume ls -q
[15113 ms] Stop (49 ms): Run: docker volume ls -q
[15113 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[15161 ms] Stop (48 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[15171 ms] Start: Run: docker ps -q -a
[15218 ms] Stop (47 ms): Run: docker ps -q -a
[15218 ms] Start: Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[15267 ms] Stop (49 ms): Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[16907 ms] Start: Run: docker volume ls -q -f dangling=true -f label=vsch.local.repository
[16953 ms] Stop (46 ms): Run: docker volume ls -q -f dangling=true -f label=vsch.local.repository
[16954 ms] Start: Run: docker volume ls -q -f dangling=true -f label=dev.container.volume
[17000 ms] Stop (46 ms): Run: docker volume ls -q -f dangling=true -f label=dev.container.volume
[17001 ms] Start: Run: docker inspect --type volume ubuntu
[17049 ms] Stop (48 ms): Run: docker inspect --type volume ubuntu
[19818 ms] Start: Run: docker volume rm ubuntu
[19866 ms] Start: Run: docker ps -q -a
[19867 ms] Start: Run: docker volume ls -q
[19867 ms] Stop (49 ms): Run: docker volume rm ubuntu
[19916 ms] Stop (50 ms): Run: docker ps -q -a
[19917 ms] Start: Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[19941 ms] Stop (74 ms): Run: docker volume ls -q
[19941 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[19968 ms] Stop (51 ms): Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[19991 ms] Stop (50 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[25471 ms] Start: Run: docker version --format {{json .}}
[25516 ms] {"Client":{"Version":"25.0.4-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.44","GitCommit":"c4cd0a9","GoVersion":"go1.21.8","Os":"windows","Arch":"amd64","BuildTime":"Fri Mar  8 09:11:41 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.3","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Wed Mar  6 12:07:44 2024","Experimental":"false","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.3","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"f417435e5f6216828dec57958c490c4f8bae4f98","GoVersion":"go1.21.8","Os":"linux","Arch":"amd64","KernelVersion":"5.15.146.1-microsoft-standard-WSL2","BuildTime":"2024-03-06T12:07:44.000000000+00:00"}}
[25518 ms] Stop (47 ms): Run: docker version --format {{json .}}
[25519 ms] Start: Run: docker pull alpine:3.16.3
[25682 ms] Start: Run: docker volume ls -q
[25728 ms] Stop (46 ms): Run: docker volume ls -q
[29485 ms] Start: Run: docker volume create --label dev.container.volume=true ubuntu
[29535 ms] Start: Run: docker volume ls -q
[29535 ms] Start: Run: docker ps -q -a
[29537 ms] Stop (52 ms): Run: docker volume create --label dev.container.volume=true ubuntu
[29537 ms] Start: Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[29592 ms] Stop (57 ms): Run: docker ps -q -a
[29592 ms] Start: Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[29593 ms] Stop (58 ms): Run: docker volume ls -q
[29593 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[29645 ms] Stop (53 ms): Run: docker inspect --type container 0de6f7091fdb 00bf97e35d6e
[29671 ms] Stop (78 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[29835 ms] Start: Run: docker volume ls -q
[29858 ms] Start: Run: docker ps -q -a
[29859 ms] Stop (322 ms): Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[29887 ms] Stop (52 ms): Run: docker volume ls -q
[29888 ms] Start: Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[29914 ms] Stop (56 ms): Run: docker ps -q -a
[29914 ms] Start: Run: docker inspect --type container 58ce29a4e35a 0de6f7091fdb 00bf97e35d6e
[29943 ms] Stop (55 ms): Run: docker inspect --type volume boot-dev-git-main-5c8fc1b41efd1bce37faf0aed5b629d6356124d15b6aea60f137ab4fdd40ae3e ubuntu vscode vscode-remote-try-go-9ad62b75d42a4d304eebbd6de726a14f22827dc2ec6163258a0723c7e3348020
[29959 ms] Preparing to apply Template 'ghcr.io/devcontainers/templates/ubuntu:latest'
[29960 ms] Running Dev Containers CLI:   templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1717889153929 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1717889153929 --log-level trace
[29963 ms] ProxyResolver#loadSystemCertificates count
[29966 ms] ProxyResolver#loadSystemCertificates count filtered
[29966 ms] Start: Run: C:\Users\evilh\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\evilh\.vscode\extensions\ms-vscode-remote.remote-containers-0.371.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1717889153929 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1717889153929 --log-level trace
[29975 ms] Stop (61 ms): Run: docker inspect --type container 58ce29a4e35a 0de6f7091fdb 00bf97e35d6e
[30158 ms] [2024-06-08T23:25:54.128Z] @devcontainers/cli 0.63.0. Node.js v20.9.0. win32 10.0.22631 x64.
[30159 ms] [2024-06-08T23:25:54.128Z] > input: ghcr.io/devcontainers/templates/ubuntu:latest
[30159 ms] [2024-06-08T23:25:54.129Z] >
[30159 ms] [2024-06-08T23:25:54.129Z] > resource: ghcr.io/devcontainers/templates/ubuntu
[30159 ms] [2024-06-08T23:25:54.129Z] > id: ubuntu
[30159 ms] [2024-06-08T23:25:54.129Z] > owner: devcontainers
[30159 ms] [2024-06-08T23:25:54.129Z] > namespace: devcontainers/templates
[30159 ms] [2024-06-08T23:25:54.129Z] > registry: ghcr.io
[30160 ms] [2024-06-08T23:25:54.129Z] > path: devcontainers/templates/ubuntu
[30160 ms] [2024-06-08T23:25:54.129Z] >
[30160 ms] [2024-06-08T23:25:54.129Z] > version: latest
[30160 ms] [2024-06-08T23:25:54.129Z] > tag?: latest
[30161 ms] [2024-06-08T23:25:54.129Z] > digest?: undefined
[30161 ms] [2024-06-08T23:25:54.129Z] > input: ghcr.io/devcontainers/templates/ubuntu:latest
[30161 ms] [2024-06-08T23:25:54.129Z] >
[30161 ms] [2024-06-08T23:25:54.129Z] > resource: ghcr.io/devcontainers/templates/ubuntu
[30162 ms] [2024-06-08T23:25:54.129Z] > id: ubuntu
[30162 ms] [2024-06-08T23:25:54.129Z] > owner: devcontainers
[30162 ms] [2024-06-08T23:25:54.129Z] > namespace: devcontainers/templates
[30162 ms] [2024-06-08T23:25:54.129Z] > registry: ghcr.io
[30162 ms] [2024-06-08T23:25:54.129Z] > path: devcontainers/templates/ubuntu
[30162 ms] [2024-06-08T23:25:54.129Z] >
[30162 ms] [2024-06-08T23:25:54.129Z] > version: latest
[30162 ms] [2024-06-08T23:25:54.129Z] > tag?: latest
[30162 ms] [2024-06-08T23:25:54.129Z] > digest?: undefined
[30163 ms] [2024-06-08T23:25:54.129Z] manifest url: https://ghcr.io/v2/devcontainers/templates/ubuntu/manifests/latest
[30165 ms] [2024-06-08T23:25:54.135Z] Loading 50 extra certificates from C:\Users\evilh\AppData\Local\Temp\vsch\certificates-9b3207a1811c345e81b95e721f2c3bebc32924a0050cf938fcb28057258a1684.pem.
[30316 ms] [2024-06-08T23:25:54.286Z] [httpOci] Attempting to authenticate via 'Bearer' auth.
[30317 ms] [2024-06-08T23:25:54.288Z] [httpOci] Invoking credsStore credential helper 'wincred'
[30317 ms] [2024-06-08T23:25:54.288Z] Start: Run: docker-credential-wincred get
[30344 ms] [2024-06-08T23:25:54.315Z] Stop (27 ms): Run: docker-credential-wincred get
[30345 ms] [2024-06-08T23:25:54.315Z] [httpOci] Failed to query for 'ghcr.io' credential from 'docker-credential-wincred': [object Object]
[30345 ms] [2024-06-08T23:25:54.315Z] [httpOci] No authentication credentials found for registry 'ghcr.io' via docker config or credential helper.
[30345 ms] [2024-06-08T23:25:54.315Z] [httpOci] No authentication credentials found for registry 'ghcr.io'. Accessing anonymously.
[30345 ms] [2024-06-08T23:25:54.315Z] [httpOci] Attempting to fetch bearer token from:  https://ghcr.io/token?service=ghcr.io&scope=repository:devcontainers/templates/ubuntu:pull
[30607 ms] [2024-06-08T23:25:54.577Z] [httpOci] 200 on reattempt after auth: https://ghcr.io/v2/devcontainers/templates/ubuntu/manifests/latest
[30607 ms] [2024-06-08T23:25:54.577Z] blob url: https://ghcr.io/v2/devcontainers/templates/ubuntu/blobs/sha256:61ed91d23448484ccf23eae27ac81cdba43bf28ee194b489c9fe02c90b22c8db
[30607 ms] [2024-06-08T23:25:54.578Z] [httpOci] Applying cachedAuthHeader for registry ghcr.io...
[30874 ms] [2024-06-08T23:25:54.844Z] [httpOci] 200 (Cached): https://ghcr.io/v2/devcontainers/templates/ubuntu/blobs/sha256:61ed91d23448484ccf23eae27ac81cdba43bf28ee194b489c9fe02c90b22c8db
[30882 ms] [2024-06-08T23:25:54.852Z] ./ : Directory
[30882 ms] [2024-06-08T23:25:54.853Z] ./.devcontainer/ : Directory
[30883 ms] [2024-06-08T23:25:54.853Z] ./.github/ : Directory
[30883 ms] [2024-06-08T23:25:54.853Z] Skipping file './NOTES.md' during blob extraction
[30883 ms] [2024-06-08T23:25:54.853Z] Skipping file './README.md' during blob extraction
[30883 ms] [2024-06-08T23:25:54.853Z] Skipping file './devcontainer-template.json' during blob extraction
[30883 ms] [2024-06-08T23:25:54.853Z] ./.devcontainer/devcontainer.json : File
[30883 ms] [2024-06-08T23:25:54.853Z] ./.github/dependabot.yml : File
[30887 ms] [2024-06-08T23:25:54.858Z] Files extracted from blob: ./.devcontainer/devcontainer.json, ./.github/dependabot.yml
[30889 ms] [2024-06-08T23:25:54.859Z] Found metadata file 'devcontainer-template.json' in blob
[30890 ms] [2024-06-08T23:25:54.860Z] Using default value for imageVariant --> jammy
[30890 ms] [2024-06-08T23:25:54.860Z] Scanning file './.devcontainer/devcontainer.json'
[30891 ms] [2024-06-08T23:25:54.861Z] Replacing imageVariant with jammy
[30891 ms] [2024-06-08T23:25:54.862Z] Scanning file './.github/dependabot.yml'
[30900 ms] Stop (934 ms): Run: C:\Users\evilh\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\evilh\.vscode\extensions\ms-vscode-remote.remote-containers-0.371.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\evilh\AppData\Local\Temp\tmp-output-dir-1717889153929 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\evilh\AppData\Local\Temp\tmp-dir-1717889153929 --log-level trace
[30900 ms] {
    "files": [
        "./.devcontainer/devcontainer.json",
        "./.github/dependabot.yml"
    ]
}
[30900 ms] Dev Containers CLI completed
[30901 ms] Copying file './.devcontainer/devcontainer.json'
[34531 ms] Stop (9012 ms): Run: docker pull alpine:3.16.3
```

Also I connected to the bootstrap container and I can see the volume is mounted with the devcontainer data, it just appears to be not be switching from the bootstrap to the actual container.

```
~ # tree -a /workspaces
/workspaces
└── ubuntu
    └── .devcontainer
        └── devcontainer.json
```

```json
# cat ubuntu/.devcontainer/devcontainer.json
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/ubuntu
{
        "name": "Ubuntu",
        // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
        "image": "mcr.microsoft.com/devcontainers/base:jammy"

        // Features to add to the dev container. More info: https://containers.dev/features.
        // "features": {},

        // Use 'forwardPorts' to make a list of ports inside the container available locally.
        // "forwardPorts": [],

        // Use 'postCreateCommand' to run commands after the container is created.
        // "postCreateCommand": "uname -a",

        // Configure tool-specific properties.
        // "customizations": {},

        // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
        // "remoteUser": "root"
}
```

**Comment by choigawoon** [comment]
having same issue, and i want a solution for this.

**Comment by chrmarti** [comment]
@evilhamsterman Thanks, what's interesting is that it doesn't move on to copy the second file `dependabot.yml` and that `docker pull alpine:3.16.3` finishes only after copying has already started. Not sure if these two are related at all, but the `alpine:3.16.3` image is used for the temporary container to copy the files over, so the image should have been pulled by the time the copying starts.

I will let it wait for `docker pull alpine:3.16.3` to finish before starting the temporary container hoping that this will improve things.

@choigawoon Could you share your log with trace logging enabled for Dev Containers (see [above](https://github.com/microsoft/vscode-remote-release/issues/9892#issuecomment-2116941175))?

**Comment by chrmarti** [comment]
@evilhamsterman Could you give Dev Containers 0.373.0-pre-release a try? That waits for the `docker pull` to finish before creating the container.

**Comment by evilhamsterman** [comment]
> @evilhamsterman Could you give Dev Containers 0.373.0-pre-release a try? That waits for the `docker pull` to finish before creating the container.

Same result hangs at copying the devcontainer.json. I don't think the docker pull is the issue, I thought it was having problems pulling all together so I pulled it myself prior to trying to make sure it was cached. 

```
[2966 ms] Dev Containers 0.373.0 in VS Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d).
[2966 ms] Initializing configuration support...
[2967 ms] Internal initialization of dev container support package...
[42302 ms] Start: Run: docker version --format {{json .}}
[42556 ms] {"Client":{"Version":"26.1.0-rd","ApiVersion":"1.44","DefaultAPIVersion":"1.45","GitCommit":"cca8e72","GoVersion":"go1.21.9","Os":"windows","Arch":"amd64","BuildTime":"Tue Apr 23 16:26:59 2024","Context":"default"},"Server":{"Platform":{"Name":""},"Components":[{"Name":"Engine","Version":"25.0.5","Details":{"ApiVersion":"1.44","Arch":"amd64","BuildTime":"Sun May 12 07:25:43 2024","Experimental":"false","GitCommit":"e63daec8672d77ac0b2b5c262ef525c7cf17fd20","GoVersion":"go1.21.10","KernelVersion":"5.15.153.1-microsoft-standard-WSL2","MinAPIVersion":"1.24","Os":"linux"}},{"Name":"containerd","Version":"v1.7.10","Details":{"GitCommit":"4e1fe7492b9df85914c389d1f15a3ceedbb280ac"}},{"Name":"runc","Version":"1.1.12","Details":{"GitCommit":"51d5e94601ceffbbd85688df1c928ecccbfa4685"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":""}}],"Version":"25.0.5","ApiVersion":"1.44","MinAPIVersion":"1.24","GitCommit":"e63daec8672d77ac0b2b5c262ef525c7cf17fd20","GoVersion":"go1.21.10","Os":"linux","Arch":"amd64","KernelVersion":"5.15.153.1-microsoft-standard-WSL2","BuildTime":"2024-05-12T07:25:43.000000000+00:00"}}
[42559 ms] Stop (257 ms): Run: docker version --format {{json .}}
[42562 ms] Start: Run: docker pull alpine:3.16.3
[42956 ms] Start: Run: docker volume ls -q
[43220 ms] Stop (264 ms): Run: docker volume ls -q
[45487 ms] Stop (2925 ms): Run: docker pull alpine:3.16.3
[47202 ms] Start: Run: docker volume create --label dev.container.volume=true ubuntu
[47484 ms] Stop (282 ms): Run: docker volume create --label dev.container.volume=true ubuntu
[47485 ms] Start: Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[48268 ms] Stop (783 ms): Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces alpine:3.16.3 sleep infinity
[48557 ms] Preparing to apply Template 'ghcr.io/devcontainers/templates/ubuntu:latest'
[48559 ms] Running Dev Containers CLI:   templates apply --workspace-folder C:\Users\dmills\AppData\Local\Temp\tmp-output-dir-1718399117918 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\dmills\AppData\Local\Temp\tmp-dir-1718399117918 --log-level trace
[48565 ms] ProxyResolver#loadSystemCertificates count
[48569 ms] ProxyResolver#loadSystemCertificates count filtered
[48571 ms] Start: Run: C:\Users\dmills\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\dmills\.vscode\extensions\ms-vscode-remote.remote-containers-0.373.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\dmills\AppData\Local\Temp\tmp-output-dir-1718399117918 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\dmills\AppData\Local\Temp\tmp-dir-1718399117918 --log-level trace
[49320 ms] [2024-06-14T21:05:18.679Z] @devcontainers/cli 0.64.0. Node.js v20.9.0. win32 10.0.22631 x64.
[49321 ms] [2024-06-14T21:05:18.679Z] > input: ghcr.io/devcontainers/templates/ubuntu:latest
[49321 ms] [2024-06-14T21:05:18.681Z] >
[49321 ms] [2024-06-14T21:05:18.681Z] > resource: ghcr.io/devcontainers/templates/ubuntu
[49321 ms] [2024-06-14T21:05:18.681Z] > id: ubuntu
[49321 ms] [2024-06-14T21:05:18.681Z] > owner: devcontainers
[49321 ms] [2024-06-14T21:05:18.681Z] > namespace: devcontainers/templates
[49321 ms] [2024-06-14T21:05:18.681Z] > registry: ghcr.io
[49322 ms] [2024-06-14T21:05:18.681Z] > path: devcontainers/templates/ubuntu
[49322 ms] [2024-06-14T21:05:18.681Z] >
[49322 ms] [2024-06-14T21:05:18.681Z] > version: latest
[49322 ms] [2024-06-14T21:05:18.681Z] > tag?: latest
[49322 ms] [2024-06-14T21:05:18.681Z] > digest?: undefined
[49322 ms] [2024-06-14T21:05:18.681Z] > input: ghcr.io/devcontainers/templates/ubuntu:latest
[49322 ms] [2024-06-14T21:05:18.681Z] >
[49325 ms] [2024-06-14T21:05:18.681Z] > resource: ghcr.io/devcontainers/templates/ubuntu
[49325 ms] [2024-06-14T21:05:18.681Z] > id: ubuntu
[49325 ms] [2024-06-14T21:05:18.681Z] > owner: devcontainers
[49325 ms] [2024-06-14T21:05:18.681Z] > namespace: devcontainers/templates
[49325 ms] [2024-06-14T21:05:18.681Z] > registry: ghcr.io
[49325 ms] [2024-06-14T21:05:18.681Z] > path: devcontainers/templates/ubuntu
[49326 ms] [2024-06-14T21:05:18.682Z] >
[49326 ms] [2024-06-14T21:05:18.682Z] > version: latest
[49326 ms] [2024-06-14T21:05:18.682Z] > tag?: latest
[49326 ms] [2024-06-14T21:05:18.682Z] > digest?: undefined
[49326 ms] [2024-06-14T21:05:18.682Z] manifest url: https://ghcr.io/v2/devcontainers/templates/ubuntu/manifests/latest
[49329 ms] [2024-06-14T21:05:18.690Z] Loading 45 extra certificates from C:\Users\dmills\AppData\Local\Temp\vsch\certificates-1e32109d88a6548321eb9924055986ce71db7bb36fcf9679b22e329d9d4ba374.pem.
[49494 ms] [2024-06-14T21:05:18.855Z] [httpOci] Attempting to authenticate via 'Bearer' auth.
[49496 ms] [2024-06-14T21:05:18.857Z] [httpOci] Invoking credsStore credential helper 'wincred'
[49496 ms] [2024-06-14T21:05:18.857Z] Start: Run: docker-credential-wincred get
[49621 ms] [2024-06-14T21:05:18.982Z] Stop (125 ms): Run: docker-credential-wincred get
[49622 ms] [2024-06-14T21:05:18.982Z] [httpOci] Attempting to fetch bearer token from:  https://ghcr.io/token?service=ghcr.io&scope=repository:devcontainers/templates/ubuntu:pull
[49941 ms] [2024-06-14T21:05:19.298Z] [httpOci] 200 on reattempt after auth: https://ghcr.io/v2/devcontainers/templates/ubuntu/manifests/latest
[49942 ms] [2024-06-14T21:05:19.298Z] blob url: https://ghcr.io/v2/devcontainers/templates/ubuntu/blobs/sha256:61ed91d23448484ccf23eae27ac81cdba43bf28ee194b489c9fe02c90b22c8db
[49942 ms] [2024-06-14T21:05:19.299Z] [httpOci] Applying cachedAuthHeader for registry ghcr.io...
[50272 ms] [2024-06-14T21:05:19.632Z] [httpOci] 200 (Cached): https://ghcr.io/v2/devcontainers/templates/ubuntu/blobs/sha256:61ed91d23448484ccf23eae27ac81cdba43bf28ee194b489c9fe02c90b22c8db
[50276 ms] [2024-06-14T21:05:19.637Z] ./ : Directory
[50277 ms] [2024-06-14T21:05:19.638Z] ./.devcontainer/ : Directory
[50278 ms] [2024-06-14T21:05:19.638Z] ./.github/ : Directory
[50278 ms] [2024-06-14T21:05:19.639Z] Skipping file './NOTES.md' during blob extraction
[50278 ms] [2024-06-14T21:05:19.639Z] Skipping file './README.md' during blob extraction
[50279 ms] [2024-06-14T21:05:19.639Z] Skipping file './devcontainer-template.json' during blob extraction
[50279 ms] [2024-06-14T21:05:19.639Z] ./.devcontainer/devcontainer.json : File
[50279 ms] [2024-06-14T21:05:19.639Z] ./.github/dependabot.yml : File
[50286 ms] [2024-06-14T21:05:19.647Z] Files extracted from blob: ./.devcontainer/devcontainer.json, ./.github/dependabot.yml
[50290 ms] [2024-06-14T21:05:19.650Z] Found metadata file 'devcontainer-template.json' in blob
[50292 ms] [2024-06-14T21:05:19.651Z] Using default value for imageVariant --> jammy
[50292 ms] [2024-06-14T21:05:19.651Z] Scanning file './.devcontainer/devcontainer.json'
[50292 ms] [2024-06-14T21:05:19.652Z] Replacing imageVariant with jammy
[50293 ms] [2024-06-14T21:05:19.653Z] Scanning file './.github/dependabot.yml'
[50307 ms] Stop (1736 ms): Run: C:\Users\dmills\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\dmills\.vscode\extensions\ms-vscode-remote.remote-containers-0.373.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\dmills\AppData\Local\Temp\tmp-output-dir-1718399117918 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\dmills\AppData\Local\Temp\tmp-dir-1718399117918 --log-level trace
[50307 ms] {
    "files": [
        "./.devcontainer/devcontainer.json",
        "./.github/dependabot.yml"
    ]
}
[50308 ms] Dev Containers CLI completed
[50308 ms] Copying file './.devcontainer/devcontainer.json'
```

**Comment by evilhamsterman** [comment]
I tried again, little different output

```
[44142 ms] [2024-06-14T21:21:00.458Z] Scanning file './.github/dependabot.yml'
[44155 ms] Stop (2201 ms): Run: C:\Users\dmills\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\Users\dmills\.vscode\extensions\ms-vscode-remote.remote-containers-0.373.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\dmills\AppData\Local\Temp\tmp-output-dir-1718400058249 --template-id ghcr.io/devcontainers/templates/ubuntu:latest --template-args {} --features [] --tmp-dir C:\Users\dmills\AppData\Local\Temp\tmp-dir-1718400058249 --log-level trace
[44156 ms] {
    "files": [
        "./.devcontainer/devcontainer.json",
        "./.github/dependabot.yml"
    ]
}
[44156 ms] Dev Containers CLI completed
[44156 ms] Copying file './.devcontainer/devcontainer.json'
[62053 ms] Start: Run: docker inspect --type container 76936c9aca25ed66b273bab4fb75c304886e98eaa846aaf31730b8dff21229c4
[62369 ms] Stop (316 ms): Run: docker inspect --type container 76936c9aca25ed66b273bab4fb75c304886e98eaa846aaf31730b8dff21229c4
```

I checked and the file has copied it seems it just doesn't want to start the next step

```
❯ docker exec keen_mahavira cat /workspaces/ubuntu/.devcontainer/devcontainer.json
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/ubuntu
{
        "name": "Ubuntu",
        // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
        "image": "mcr.microsoft.com/devcontainers/base:jammy"

        // Features to add to the dev container. More info: https://containers.dev/features.
        // "features": {},

        // Use 'forwardPorts' to make a list of ports inside the container available locally.
        // "forwardPorts": [],

        // Use 'postCreateCommand' to run commands after the container is created.
        // "postCreateCommand": "uname -a",

        // Configure tool-specific properties.
        // "customizations": {},

        // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
        // "remoteUser": "root"
}
```

**Comment by andi-blafasl** [comment]
I'm experiencing the exact same behavior with Rancher Desktop. Switching vom Rancher to Docker Desktop, everything is working fine.

**Comment by evilhamsterman** [comment]
Still seems to be an issue, though now if I leave it running I get the repeated message 
```
[206975 ms] Dev Containers CLI completed
[206975 ms] Copying file './.devcontainer/devcontainer.json'
[229294 ms] Received message: []
[237124 ms] Extensions cache, copy to shared cache: None
[289310 ms] Received message: []
[297135 ms] Extensions cache, copy to shared cache: None
[349314 ms] Received message: []
[357142 ms] Extensions cache, copy to shared cache: None
```

running 0.381.0

**Comment by pxslip** [comment]
I recently found that enabling the option `dev.containers.executeInWSL` made this issue disappear. That may be coincidence, but wanted to share in case it helps someone else or helps identify the root cause.

**Comment by evilhamsterman** [comment]
@chrmarti can you look at it again. It's easily reproducible just use Rancher Desktop as the your docker provider

**Comment by chrmarti** [comment]
Seeing it after uninstalling Docker Desktop. I think I was using its docker executable with Rancher previously. The following reproduces the problem, it works with Docker Desktop on Mac:
```sh
docker run -d --name testcontainer alpine:latest sleep inf
echo hello | docker exec -i testcontainer sh -c 'cat >/root/world.txt' # hangs with Rancher Desktop
docker exec -i testcontainer cat /root/world.txt
```

@evilhamsterman Could you open an issue for Rancher Desktop? This seems to be specific to their build of the docker CLI. Thanks!

**Comment by evilhamsterman** [comment]
Someone already has https://github.com/rancher-sandbox/rancher-desktop/issues/7054. Did you run that on Windows, if so in PowerShell or cmd? If that's the case it looks like it's not accepting pipe input and maybe that's the problem
